### PR TITLE
cart: Add option to remove applied voucher from cart

### PR DESCRIPTION
**Pull request by carstendietrich**
This PR allows the user to remove already applied vouchers from card.

### CODE REVIEW

File: cart/infrastructure/InMemoryBehaviour.go [hunk 1]
```diff
@@ -354,6 +355,24 @@ func (cob *InMemoryBehaviour) ApplyVoucher(ctx context.Context, cart *domaincart
...
 }
 
+// RemoveVoucher removes a voucher from the cart
+func (cob *InMemoryBehaviour) RemoveVoucher(ctx context.Context, cart *domaincart.Cart, couponCode string) (*domaincart.Cart, domaincart.DeferEvents, error) {
+	for i, coupon := range cart.AppliedCouponCodes {
+		if coupon.Code == couponCode {
+			cart.AppliedCouponCodes[i] = cart.AppliedCouponCodes[len(cart.AppliedCouponCodes)-1]
+			cart.AppliedCouponCodes[len(cart.AppliedCouponCodes)-1] = domaincart.CouponCode{}
+			cart.AppliedCouponCodes = cart.AppliedCouponCodes[:len(cart.AppliedCouponCodes)-1]
+		}
```
Comment (jeinfeldt) [inline cart/infrastructure/InMemoryBehaviour.go]: We assume that only one coupon code matches, right? In that case you could insert a break after the removal of the element, as there is no need to iterate through the codes further

File: cart/module.go [hunk 2]
```diff
@@ -158,6 +158,10 @@ func (r *routes) apiRoutes(registry *web.RouterRegistry) {
 	registry.HandlePost("cart.api.applyVoucher", r.apiController.ApplyVoucherAndGetAction)
 	registry.HandlePut("cart.api.applyVoucher", r.apiController.ApplyVoucherAndGetAction)
 
+	registry.Route("/api/cart/removevoucher", `cart.api.removeVoucher(couponCode)`)
+	registry.HandlePost("cart.api.removevoucher", r.apiController.RemoveVoucherAndGetAction)
+	registry.HandleDelete("cart.api.removevoucher", r.apiController.RemoveVoucherAndGetAction)
+
```
Comment (jeinfeldt) [inline cart/module.go]: cart.api.removeVoucher is declared in camel case but used in lower case

cart.ape.removevoucher. If an exact match is necessary, please adjust

**Comment by jeinfeldt** [comment]
Thanks for contributing!

File: cart/module.go (hunk 2 above)
Comment (carstendietrich) [reply cart/module.go]: I updated the handle name accordingly.

File: cart/infrastructure/InMemoryBehaviour.go (hunk 1 above)
Comment (carstendietrich) [reply cart/infrastructure/InMemoryBehaviour.go]: Good point, I added the break.

**Comment by jeinfeldt** [comment]
Thanks for contributing! 👍

**Pull request merged**